src/uct/ib/base/ib_md.c

```diff
@@ -309,13 +310,40 @@ uint8_t uct_ib_md_get_atomic_mr_id(uct_ib_md_t *md)
 #endif
 }
 
+static void uct_ib_md_print_mem_reg_err_msg(ucs_log_level_t level, void *address,
+                                            size_t length, uint64_t exp_access,
+                                            int exp)
+{
+    char *str = ucs_alloca(32);
+    char msg[400] = {0};
+
+    if (exp) {
+        ucs_snprintf_zero(msg, sizeof(msg),
+                          "ibv_exp_reg_mr(address=%p, length=%zu, exp_access=0x%lx) failed: %m. ",
+                          address, length, exp_access);
+    } else {
+        ucs_snprintf_zero(msg, sizeof(msg),
+                          "ibv_reg_mr(address=%p, length=%zu, access=0x%lx) failed: %m. ",
+                          address, length, exp_access);
+    }
+
+    str = ucs_sys_get_mem_lock_limit(str, 32);
+    if ((strcmp(str, "'unlimited'")) && (strcmp(str, "<unknown>"))) {
```

Comment:
it's not very nice to compare strings to check for specific situation, better use return codes (int/enum/status)
for this reason looks like it's better to just to "getrlimit" in this function, without the UCS wrapper.